function.k

```diff
@@ -373,18 +459,27 @@ rule <k> pcsContractPart(C:Id, function F:Id (P1:FParameters) .FunQuantifiers
                <ctname> Cn:Id </ctname>
 	       ...
            </contractInstance> 
+
+  */   
+    
+      rule <k> functionCall(Sender:Int; C:Int ; F:Id ; Es:Values ; M:Msg) => createTransaction(.List) ~> functionCall(F;Es) ~> returnContext(C) ... </k>
+        <contractStack>  (.List => ListItem(C))  ListItem(-1)   </contractStack>
+         <Msg> M1=>M </Msg>
+         <MsgStack> (.List => ListItem(M1)) ... </MsgStack>
+         <functionStack> (.List => ListItem(F)) ... </functionStack>
        
-       rule <k> functionCall(Sender:Int; C:Int ; F:Id ; Es:Values ; M:Msg); => FunCall(C:Int, F:Id, Es:Values,M:Msg) ~> . ... </k>
+       
+       rule <k> functionCall(Sender:Int; C:Int ; F:Id ; Es:Values ; M:Msg) => /*FunCall(C:Int, F:Id, Es:Values,M:Msg)*/ createTransaction(L:List) ~> functionCall(F;Es) ~> returnContext(C)  ... </k>
         <contractStack>  (.List => ListItem(C)) L:List ListItem(-1)   </contractStack>
```

Comment:
As I pointed out before for the `CallConstructor` rule, this should be refactored so that we do not have to duplicate the rule in order to distinguish between the "empty" `contractStack` and a non-empty one (modulo the `-1` value which is in it from the start). 

Besides, those two rules are technically "incorrect" because there is no side-condition specifying that `L` must be non-empty, so this rule could potentially always match. The reason things work is because of the order in which the rules are written in the file, but this should be regarded as an implementation detail which we should not rely on for obtaining the desired behaviour. 

If you look on the Trello board you will see a card for `CallConstructor` where I explain the issue in detail. This case (`functionCall`) is very similar and should be handled in the same way.